src/codegen/utilities/interpreter_helpers.py

```diff
@@ -322,9 +322,19 @@ def get_c_function_call_template(func):
     return "/default_c_function_call.py.mako"
 
 
-def get_callback_param_data_types(params):
+def get_callback_param_data_types(func_params):
     """Gets the data types for call back function parameters."""
-    return [p["ctypes_data_type"] for p in params]
+    callback_func_param = next(p for p in func_params if p.parameter_name == "callback_function")
+    callback_data_param = next(p for p in func_params if p.parameter_name == "callback_data")
+    callback_param_types = []
+    # callback_param_types = [result_type, [**ctypes_data_type** of callback_params],
+    # **ctypes_data_type** of callback_data_param]
+    callback_param_types.append("ctypes.c_int32")
+    callback_param_types.extend(
+        [p["ctypes_data_type"] for p in callback_func_param.callback_params]
+    )
+    callback_param_types.append(callback_data_param.ctypes_data_type)
+    return callback_param_types
```

Comment:
I suspect that this is what you were trying to go for:
```suggestion
    return (
        ["ctypes.c_int32"] +
        [p["ctypes_data_type"] for p in callback_func_param.callback_params] +
        [callback_data_param.ctypes_data_type]
    )
```

Reply:
Updated.